Sources/SwiftDocCUtilities/ArgumentParsing/Options/Source Repository/SourceRepositoryArguments.swift

```diff
@@ -0,0 +1,91 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2022 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import ArgumentParser
+import Foundation
+import SwiftDocC
+
+/// Command-line arguments for specifying the catalog's source repository information.
+public struct SourceRepositoryArguments: ParsableArguments {
+    public init() {}
+    
+    /// The root path on disk of the repository's checkout.
+    ///
+    /// By default, this is set to the current working directory.
+    @Option(
+        help: ArgumentHelp(
+            "The root path on disk of the repository's checkout.",
+            discussion: "By default, this is set to the current working directory."
+        )
+    )
+    public var checkoutPath: String?
```

Comment:
I'm a little concerned about this... Seems like the possibility of accidentally breaking this and being confused about why things aren't working is pretty high. Maybe we can help address this with specific support in the Swift-DocC Plugin though?

Reply:
Just occurred to me- maybe we can at least validate that the path given here is the root of a git repository by looking for a `.git` directory? Not sure if there's a more robust way to ask `git` if a given path is the root of a git repo...

...

After some [searching](https://stackoverflow.com/questions/957928/is-there-a-way-to-get-the-git-root-directory-in-one-command), I found this as well which might be helpful to use as a better default instead of the current working directory:

```
git rev-parse --show-toplevel
```

Reply:
I would prefer not adding `git` as a dependency to docc. I think higher-level build systems—which may already have some VSC integration—should determine the appropriate path.

Reply:
That seems fine- although just checking for the `.git` and emitting a warning if we don't find it doesn't seem like _too_ much of an overstep for `docc` to do.

I think you're right though that generally we should have higher-level build systems handle this though. In that case- maybe we should remove the default value and require it be explicitly passed? I think users won't expect that this is a requirement for this feature at all and will be confused when it breaks. Requiring for direct use of `docc` and then adding support in the DocC-Plugin to pass a valid value by default seems like a better route here.

Reply:
Based on past experience I would advise against trying to insect git repositories if possible. If we do this the implementation should document all assumptions, no matter how silly they may seem, and offer options for developers skips these checks. One basic assumption that's easy to overlook is that ".git" is a directory within the main repo but it's a file in submodule checkouts and worktree checkouts.

Reply:
Ah got it. Okay I'll definitely reach out to when we're looking at doing this in the DocC-Plugin then. Since there are so many potential pitfalls here it seems like removing the responsibility from DocC is probably the right move.

Reply:
For the plugin, I'm wondering if some of the information we need can be provided by SwiftPM directly, since it already works with git.